plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/IterateMediatorExtFactory.java

```diff
@@ -67,26 +73,37 @@ protected Mediator createSpecificMediator(OMElement omElement) {
 	OMAttribute expression = omElement.getAttribute(ATT_EXPRN);
 	if (expression != null) {
 	    try {
-		((IterateMediator) mediator).setExpression(SynapseXPathFactory.getSynapseXPath(omElement, ATT_EXPRN));
+		((IterateMediator) mediator).setExpression(SynapsePathFactory.getSynapsePath(omElement, ATT_EXPRN));
 	    } catch (JaxenException e) {
-		// ignore
+	    	// ignore
 	    }
 	}
 
 	OMAttribute attachPath = omElement.getAttribute(ATT_ATTACHPATH);
-	String attachPathValue = ".";
-	if (attachPath != null) {
-	    attachPathValue = attachPath.getAttributeValue();
-	}
-
+	
 	try {
-	    SynapseXPath xp = new SynapseXPath(attachPathValue);
-	    Log log = LogFactory.getLog(IterateMediatorFactory.class);
-	    OMElementUtils.addNameSpaces(xp, omElement, log);
-	    ((IterateMediator) mediator).setAttachPath(xp);
-	} catch (JaxenException e) {
-	    // ignore
-	}
+        SynapsePath attachSynapsePath;
+
+        if (attachPath != null) {
+            attachSynapsePath = SynapsePathFactory.getSynapsePath(omElement, ATT_ATTACHPATH);
+            ((IterateMediator)mediator).setAttachPathPresent(true);
+
+        } else {
+            if (((IterateMediator)mediator).getExpression() instanceof  SynapseJsonPath){
+                attachSynapsePath = new SynapseJsonPath(DEFAULT_JSON_ATTACHPATH);
+            } else {
+                attachSynapsePath = new SynapseXPath(DEFAULT_XML_ATTACHPATH);
+            }
+            ((IterateMediator)mediator).setAttachPathPresent(false);
+        }
+        
+        Log log = LogFactory.getLog(IterateMediatorFactory.class);
+        OMElementUtils.addNameSpaces(attachSynapsePath, omElement, log);
+        ((IterateMediator) mediator).setAttachPath(attachSynapsePath);
+
+    } catch (JaxenException e) {
+		//ignore
```

Comment:
Is ignoring the exception here correct?

Reply:
Yes, In dev studio, these try block checks are performed for validation purposes only.